### DISCUSSION TITLE
[draft-v1] Figure 3: Increase of Software Engineers

### DISCUSSION THREAD

**Discussion by pllim** [astropy/astropy-v5.0-paper#49] [Figure comments] [answered]
The most obvious trend that I can see is the increasing reliance on Software Engineers. From personal experience, these engineers are usually tied to a certain institution and motivated by their mission goals (e.g., HST, JWST, DKIST, NoirLab, VRO, etc). Just securing our own soft funding is insufficient to keep them around; We need to lobby or work with those institution to ensure their continued support as well.

Some of the students might have also shifted to software engineers over time, but the numbers do not match, which suggests that most software engineers joined as part of their day jobs.

The skillsets that software engineers bring are essential for the project, especially given the increased code and infrastructure complexity over time.

**Comment by pllim** [comment]
There are also rare cases where a software engineer status changes back to a student as the person pursues tertiary education.

**Comment by hamogu** [comment]
Numbers are small though and the ">50 commits" requirement excludes those that just implement a could of features. Looking at the list of names on the spreadsheet, almost all the software engineers are from STScI, so really that's just that STScI contributes with more people. (There are notable exceptions, but those are very small numbers.). Also, why is Tom A listed as a software engineer when is role is listed as "operations scientist" in the same table?

**Reply by adrn** [reply]
What category are you proposing Tom A should be in? @taldcroft should also advise here. My current (broad and imperfect) classes are: student, research staff/faculty, and software engineer. For "research staff/faculty", I was thinking that this would include people for which research articles and advising students in research projects are primary deliverables of their job. For "software engineering", I used this for people where software or support products are primary deliverables.

**Reply by adrn** [reply]
Please open an issue (@taldcroft or @hamogu) about updating the [data file](https://docs.google.com/spreadsheets/d/1iGiWn-KTG88lMZ6_79y1NkoK7kJ5-u5K6AajlQ8xVm0/edit#gid=1945969023) if you would like to suggest a different classification.

**Reply by taldcroft** [reply]
I'm squarely in research staff/faculty, job title "Senior Astrophysicist".

**Reply by adrn** [reply]
OK great! I'll update now

**Comment by adrn** [comment, marked as answer]
Closing this - please feel free to open an issue if anything is unresolved.
